pkgs/applications/science/math/mathematica/default.nix

```
@@ -92,6 +92,10 @@ stdenv.mkDerivation rec {
     sed -i -e 's/^PATH=/# PATH=/' MathInstaller
     sed -i -e 's/\/bin\/bash/\/bin\/sh/' MathInstaller
 
+    # Install the desktop items
+    sed -i -e 's/Executables\/Mathematica/..\/..\/bin\/mathematica/' MathInstaller
```

```suggestion
    sed -i -e 's|Executables/Mathematica|../../bin/mathematica|' MathInstaller
```

Also can change line that is above like this too.

---

Also could even replace all of the `sed` calls to `substituteInPlace` as i see.

---

The regex for `PATH` would be tricky. I'll try updating the other two to use `substituteInPlace`.

---

I've replaced the two `sed` calls with `substituteInPlace`, and also expanded on the comment.

It still builds and works as before.